packages/noir-ethereum-api/test/e2e.test.ts

```diff
@@ -55,8 +60,9 @@ describe('e2e', () => {
   })
 
   it('proof fails: invalid state root', async () => {
-    alterArray(accountWithProof.stateRoot);
-    expectCircuitFail(generate_and_verify_simple_proof(defaultTestCircuitInputParams, oracles))
+    const inputParams = clone(defaultTestCircuitInputParams)
+    alterArray(inputParams.state_root as string[]);
```

Comment:
No need for clone, simpler version:
```ts
alterArray([...inputParams.state_root] as string[]);
```

Reply:
Without clone we are altering shared state that impact other tests.`alterArray` is a mutable function with side-effects. In fact other tests are incorrectly implemented. Try to comment `alterArray` in `'proof fails: invalid address'` - test is still green because of alteration of `accountWithProof` in previous test. Added task for fixing this issue to board.